services/prefect/flows/generated_flows.py

```diff
@@ -73,72 +74,54 @@ def initialize_sentry(sentry_dsn: str):
     sched = None
     if schedule:
         sched = CronSchedule(f"{ix % 60} */4 * * *")
 
     with Flow(cls.__name__, sched) as flow:
-        connstr = EnvVarSecret("COVID_DB_CONN_URI")
-        sentry_dsn = EnvVarSecret("SENTRY_DSN")
-        sentry_sdk_task = initialize_sentry(sentry_dsn)
 
-        d = create_scraper(cls)
-        fetched = fetch(d)
-        normalized = normalize(d)
-        validated = validate(d)
-        done = put(d, connstr)
+        # check if scraper has etag checking
+        # if so and the data has not been updated since the last check, set skip_flow to true
+        skip_flow = False
+        if issubclass(cls, ETagCacheMixin):
+            skip_flow = check_if_skip_cached_flow(cls)
 
-        d.set_upstream(sentry_sdk_task)
-        normalized.set_upstream(fetched)
-        validated.set_upstream(normalized)
-        done.set_upstream(validated)
-
-    return flow
+        with case(skip_flow, False):
+            connstr = EnvVarSecret("COVID_DB_CONN_URI")
+            sentry_dsn = EnvVarSecret("SENTRY_DSN")
+            sentry_sdk_task = initialize_sentry(sentry_dsn)
 
+            d = create_scraper(cls)
+            fetched = fetch(d)
+            normalized = normalize(d)
+            validated = validate(d)
+            done = put(d, connstr)
 
-def create_cdc_single_state_flow():
-    with Flow(CDCCovidDataTracker.__name__) as flow:
-        state = prefect.Parameter("state")
-        connstr = EnvVarSecret("COVID_DB_CONN_URI")
-        sentry_dsn = EnvVarSecret("SENTRY_DSN")
-        sentry_sdk_task = initialize_sentry(sentry_dsn)
+            d.set_upstream(sentry_sdk_task)
+            normalized.set_upstream(fetched)
+            validated.set_upstream(normalized)
+            done.set_upstream(validated)
 
-        d = create_scraper(CDCCovidDataTracker, state=state)
-        fetched = fetch(d)
-        normalized = normalize(d)
-        validated = validate(d)
-        done = put(d, connstr)
-
-        d.set_upstream(sentry_sdk_task)
-        normalized.set_upstream(fetched)
-        validated.set_upstream(normalized)
-        done.set_upstream(validated)
-
-    return flow
-
-
-def create_cdc_all_states_flow(schedule=True):
-    """Creates a flow that runs the CDC data update on all states."""
-    sched = None
-    if schedule:
-        sched = CronSchedule("17 */4 * * *")
-
-    flow = Flow("CDCAllStatesDataUpdate", sched)
-    for state in ALL_STATES_PLUS_DC:
-        task = StartFlowRun(
-            flow_name=CDCCovidDataTracker.__name__,
-            project_name="can-scrape",
-            wait=True,
-            parameters={"state": state.abbr},
-        )
-        flow.add_task(task)
+        with case(skip_flow, True):
+            skip_cached_flow(cls.__name__)
 
     return flow
 
 
 def create_main_flow(flows: List[Flow], project_name):
-    schedule = CronSchedule("0 */3 * * *")
+    schedule = CronSchedule("0 */4 * * *")
```

Comment:
increase the time between MainFlow runs from 3 to 4 hours. The best practice I think would be to have the next MainFlow trigger on the completion of the previous one, but this should be a start